woocommerce/woocommerce-ios
# [Mobile Payments] Set minimum WCPay version to 4.0.0 for Canada

Closes: #6601  

Delay merge until after the release freeze of 9.0 – there is no need to rush this out in the next release, 9.1 is fine.

### Description

WCPay 4.0.0 adds support for Interac refunds, so it is required as the minimum version for IPP in Canada, as Interac is an important payment method there.

Interac payments are not supported in the US, so there is no need to update the minimum required version for US merchants yet. To allow that, this PR moves the minimum version to the `CardPresentPaymentsConfiguration`, so that it can be associated with a country and other configuration details.


### Testing instructions


The easiest way to test this is to edit the version number of the WCPay plugin on your test site. This does carry some risk of breaking your test site though! To change the version, open WPAdmin on your site, and do the following:

1. Open the `Plugin File Editor` from either the `Plugins` menu, or the `Tools` menu (it depends when you set up the test site) 
2. Select `WooCommerce Payments` or `WooCommerce Stripe Gateway` in the dropdown on the top right, then click `Select` (1️⃣ in the screenshot)
3. In the comment at the top of the main plugin file (e.g. `woocommerce-payments.php`), there's a version number (2️⃣ in the screenshot). Change this to the version you need.
4. Click `Update file` to save the version change (3️⃣ in the screenshot)

![Plugin file editor](https://user-images.githubusercontent.com/2472348/163567926-2c1503e1-7062-49b7-8414-7c6fcae8bd7c.png)

#### Testing Canada WCPay
1. On a Canada test store's WPAdmin, change the plugin version to 3.9.0
2. Open the app, and go to `Menu > ⚙️ > In-Person Payments`, making sure you see the `Connecting to your account` screen to be sure of using up-to-date data.
3. Observe that you're shown the `Unsupported WooCommerce Payments Version` screen
4. In WPAdmin, change the plugin version to 4.0.0
5. In the app, tap `Refresh After Updating`
6. Observe that you can use `Manage Card Reader`, and successfully connect to a reader again.

#### Other test versions
Repeat the tests with the following version numbers – I've checked them all, including Stripe Gateway in Canada which is not supported with any version yet.

Country | Plugin | Minimum supported version 
CA | WCPay | 4.0.0
CA | Stripe | N/A
US | WCPay | 3.2.1
US | Stripe | 6.2.0
Other | WCPay | N/A
Other | Stripe | N/A

### Screenshots



https://user-images.githubusercontent.com/2472348/163569223-ea97e8b8-592c-416e-8bbc-a45a85bd6d34.mp4


---
